### ISSUE TITLE
Uncaught TypeError: Cannot read property 'addEventListener' of null

### ISSUE THREAD

**Issue by uwolf** [wernerjoss/grav-plugin-fullcalendar#30]
Hi Werner,

I'm running Grav 1.74.7 with fullcalendar v0.2.7 and a rather exotic theme called "solarized".
Fullcalendar installation via Admin Panel
Whenever i activate the Plugin, i get the following error and my JS for site navigation and other JS on the site won't run anymore.  Do you have any ideas for a quick fix?

```
main.js:388 Uncaught TypeError: Cannot read property 'addEventListener' of null
    at listenBySelector (main.js:388)
    at Calendar.bindHandlers (main.js:6619)
    at Calendar.render (main.js:6595)
    at HTMLDocument.<anonymous> (calendar.js:244)
    at i (jquery-2.x.min.js:2)
    at Object.fireWith [as resolveWith] (jquery-2.x.min.js:2)
    at Function.ready (jquery-2.x.min.js:2)
    at HTMLDocument.J (jquery-2.x.min.js:2)
listenBySelector @ main.js:388
Calendar.bindHandlers @ main.js:6619
Calendar.render @ main.js:6595
(anonymous) @ calendar.js:244
i @ jquery-2.x.min.js:2
fireWith @ jquery-2.x.min.js:2
ready @ jquery-2.x.min.js:2
J @ jquery-2.x.min.js:2
```

**Comment by wernerjoss** [comment]
Hi @uwolf ,
this looks like a jquery issue - your theme seems to use a very ancient jquery version (1.11 - see [here](https://github.com/hughbris/grav-theme-solarize/blob/develop/js/jquery.min.js)).
my plugin assumes, jquery is loaded by the theme, so it does not do this once again on its own.
but, it has never been testet with jquery < 2.1 or so.
my best bet is, you try a more recent version of jquery,.

**Comment by wernerjoss** [comment]
.. additional note: your log indicates that jquery 2.x from the system is used, not 1.11 from the theme.
so, you could just try
`  
collections:
  jquery: 'system://assets/jquery/jquery-3.x.min.js'

`
in system.yaml.

**Comment by uwolf** [comment]
Thanks for your answer. But in this case, I'm out of luck though. The theme uses the skel library which needs jquery < 3.x and is itself discontinued. I'll try a downgrade of your plugin.

**Comment by wernerjoss** [comment]
hm,
I just installed the solarize theme here in my local docker test site, activated and checked - fullcalendar looks ugly, but it works.
and that is true with jquery 2.x AND jquery 3.x in system.yaml.
so your problem must be due another reason.
can you post your user/config/plugins/fullcalendar.yaml here ?
additionally, can you check what happens when you switch to another theme (e.g. quark) ?

**Comment by uwolf** [comment]
many thanks for your time. 
I did not do any fancy configuration yet. just wanted to make it run (again) so it just:
enabled: true 
in quark, JS works, but fc is empty, does not fetch the remote ical file (it's not google but from an holyday home rental site)

I used to have a running version, but can't remember which one (0.2.1 probably). But since then i upgraded all plugins and grav itself.

**Comment by wernerjoss** [comment]
ok, so if an older version used to work, maybe it is a config issue.
what template do you use for your calendar page ?
(should be calendar.html.twig from the plugin, older versions did not provide this)

**Comment by uwolf** [comment]
just added 
[fullcalendar icsfile="https://www.example.com/some_calendar.ics" ][/fullcalendar] to a default page so far.

**Comment by wernerjoss** [comment]
ok, then you should try the proposed template:
![grafik](https://user-images.githubusercontent.com/36265757/117697859-450ab200-b1c3-11eb-8f51-e7ed4f75aaa6.png)

**Comment by wernerjoss** [comment]
...for your remote ical file working, the CORS API Url is essential:
![grafik](https://user-images.githubusercontent.com/36265757/117699532-41782a80-b1c5-11eb-97a2-570d8dad10f2.png)

**Comment by uwolf** [comment]
Same error, but i just noticed: The error only occurs at the home page. with any other page, it works. But I have the same navigation and language switcher partials in any of these pages (including home)

Additonally this when it trys to fetch the calendar:
```
Access to XMLHttpRequest at 'undefinedhttps://www.example/some_calendar.ics' from origin 'http://my-site.de' has been blocked by CORS policy: Cross origin requests are only supported for protocol schemes: http, data, chrome, chrome-extension, chrome-untrusted, https.
4a03927d1773e64519245db441db3353.js:2 GET undefinedhttps://www.example/some_calendar.ics net::ERR_FAILED
```

**Comment by wernerjoss** [comment]
looks like CORS API Url is not set / empty - are you sure it is ?
addititonally, you can set verbose=true; in assets/calendar.js to see what is going on in the browser developer tools (hit F12)

**Comment by uwolf** [comment]
CORS is set
![Bildschirmfoto zu 2021-05-10 19-34-46](https://user-images.githubusercontent.com/6449745/117701020-0545c980-b1c7-11eb-8eeb-a12d6eb630dc.png)

But the request is "undefinedhttps://...." not "https://......" so something prepends an "undefined" to the protocol.....???

**Comment by uwolf** [comment]
Calendar URL:undefinedhttps://www.mycalsource/calfile.ics

**Comment by wernerjoss** [comment]
I have no idea where this 'undefined' comes from - please enable verbose in line 5 of calendar.js, then you should see what is going on in javascript console.
see lines 39 .. 67 in calendar.js.
what do the console.log() statements tell ?

**Comment by uwolf** [comment]
cfgfilestring: https://www.mysource.de/calfile.ics
YAML CFG File:  https://www.mysource.de/calfile.ics
cfgfiles[]: ["https://www.mysource.de/calfile.ics"]
default CORS Url: https://cors-anywhere.herokuapp.com/
CORS Url: https://cors-anywhere.herokuapp.com/
Calendar URL:undefinedhttps://www.mysource.de/calfile.ics

**Comment by wernerjoss** [comment]
ok, I see there is something not propagated correctly.
I will look into this further during the next days...

**Comment by uwolf** [comment]
thank you! meanwhile I try to find out why my JS navigation/lang switcher work on any sub page but not on the home page.

**Comment by wernerjoss** [comment]
well, it is getting a bit clearer now:
1st, there was a regression w.r. to the CORS Url evaluation block, which just had to be moved to resolve the 'undefined' issue.
after that, I was again able to get the ics data from the remote origin, but only AFTER confirmation access for development purpose via the URL copied from the javascripts console.
abviously, the herokuapp CORS proxy is no longer available for public use, see [this](https://github.com/Rob--W/cors-anywhere/issues/301) thread.
I have pushed my modifications to master, so you can experiment with it and see what happens.
of course, this is no working solution for remote ics Calendars.
ATM, I would recommend to just pull the remote ics File(s) to your own site in user/data/calendars and use that with the fc plugin, which works fine.
if you have access to cron on your server, this should be easy via wget.
I'll think some more about how to solve the CORS issue via the plugin, but if this cannot be done with reasonable effort, I will probably remove it completely.
also note there is a solution to automatically create ics files from [remote CalDav Servers](https://github.com/wernerjoss/caldav2ics)

**Comment by wernerjoss** [comment]
so, after some investigation and more or less unsuccessful tests with [free CORS proxies](https://nordicapis.com/10-free-to-use-cors-proxies/), I decided to implement a local proxy with the plugin.
it is now in master branch and works for me here, even from a local docker image pulling ics files from remote site on the web.
note that you need to adapt the CORS api url as described in the README.
as always, comments and feedback are welcome :smile:

**Comment by wernerjoss** [comment]
FYI, I'm closing this now, as I release a new version which solves the CORS proxy issue, as well as the mentioned 'undefined' problem.
The original uncaught type error is a theme issue and not related to this plugin.
if you install the new release and find any other problems, feel free to open a new issue.
